### PR TITLE
[Feature] specify sort col for materialized view

### LINKED ISSUE #20178: Specify sort key for materialized view
## Feature request

Specify the sort key for materialized view with a `order by` syntax:
```sql
CREATE MATERIALIZED VIEW mv_name
DISTRIBUTED BY HASH(guid)
ORDER BY (customer_id)
REFRESH ASYNC
AS 
(query)
```

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by huangfeng1993**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #20178

## Problem Summary(Required) ：

https://hsn4s18zbn.feishu.cn/docx/SU2zdmjkzoMXvVxEKWecIdtWn8f

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.0

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -373,36 +373,64 @@ private void genColumnAndSetIntoStmt(CreateMaterializedViewStatement statement,
...
                 }
-                if (column.getType().getPrimitiveType() == PrimitiveType.VARCHAR) {
-                    column.setIsKey(true);
-                    column.setAggregationType(null, false);
-                    theBeginIndexOfValue++;
-                    break;
+            }
+
+            if (keyCols.isEmpty()) {
+                throw new SemanticException("The number of Sort columns is 0");
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:410]: How about using the term "sort key" ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 2]
```diff
@@ -373,36 +373,64 @@ private void genColumnAndSetIntoStmt(CreateMaterializedViewStatement statement,
...
+            }
+
+            if (keyCols.size() > mvColumns.size()) {
+                throw new SemanticException("The number of Sort columns should be less than the number of columns.");
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:414]: How about using the term "sort key" ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 3]
```diff
@@ -373,36 +373,64 @@ private void genColumnAndSetIntoStmt(CreateMaterializedViewStatement statement,
...
-                        column.setAggregationType(null, false);
+            List<String> keyCols = statement.getSortKeys();
+            if (keyCols == null) {
+                keyCols = Lists.newArrayList();
+                int theBeginIndexOfValue = 0;
+                int keySizeByte = 0;
+                for (; theBeginIndexOfValue < mvColumns.size(); theBeginIndexOfValue++) {
+                    Column column = mvColumns.get(theBeginIndexOfValue);
+                    keySizeByte += column.getType().getIndexSize();
+                    if (theBeginIndexOfValue + 1 > FeConstants.SHORTKEY_MAX_COLUMN_COUNT ||
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:386]: what if user specified `order by` exceeds this limit ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java (hunk 2 above)
Comment (huangfeng1993) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:414]: ok

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java (hunk 3 above)
Comment (huangfeng1993) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:386]: it can work normally.

**Review by murphyatwork — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**